agent/src/com/cloud/agent/AgentShell.java

```diff
@@ -107,7 +108,16 @@ public String getPod() {
 
     @Override
     public String getHost() {
-        return _host;
+        String[] hosts = _host.split(",");
+        if (_hostCounter >= hosts.length) {
+            _hostCounter = 0;
+        }
+        s_logger.info("Connecting to host: " + hosts[_hostCounter % hosts.length]);
```

Comment:
I'll refactor this, however this prints the next 'host' the `Agent` class may attempt to connect to.